src/pages/student/single-course/ContactParticipants.tsx

```diff
@@ -0,0 +1,101 @@
+import { gql, useMutation, useQuery } from '@apollo/client';
+import { Button, useBreakpointValue, useTheme, useToast } from 'native-base';
+import { useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Participant } from '../../../gql/graphql';
+import SendParticipantsMessageModal from '../../../modals/SendParticipantsMessageModal';
+import { SelectParticipants } from '../../../widgets/SelectParticipants';
+
+type ContactProps = {
+    subcourseId: number;
+    refresh: () => void;
+};
+
+const ContactParticipants: React.FC<ContactProps> = ({ subcourseId, refresh }) => {
+    const toast = useToast();
+    const { t } = useTranslation();
+    const { sizes } = useTheme();
+
+    const [showMessageModal, setShowMessageModal] = useState(false);
+
+    const { data: participantsData } = useQuery(
+        gql(`
+        query SubcourseParticipantsForContact($subcourseId: Int!) {
+            subcourse(subcourseId: $subcourseId) {
+                participants { 
+                    id
+                    firstname
+                    lastname
+                }
+            }
+        }
+    `),
+        { variables: { subcourseId: subcourseId } }
+    );
+
+    const [sendMessage, _sendMessage] = useMutation(
+        gql(`
+            mutation sendMessage($subject: String!, $message: String!, $subcourseId: Int!, $participants: [Int!]!) {
+                subcourseNotifyParticipants(subcourseId: $subcourseId, title: $subject, body: $message, participantIDs: $participants, fileIDs: [])
+            }
+        `)
+    );
+
+    const [selectedParticipants, setSelectedParticipants] = useState<number[]>([]);
+
+    const ButtonContainer = useBreakpointValue({
+        base: '100%',
+        lg: sizes['desktopbuttonWidth'],
+    });
+
+    const onSendMessage = useCallback(
+        async (subject: string, message: string) => {
+            if (subject && message && participantsData) {
```

Comment:
I honestly like it the other way around - but that's just a suggestion.
```suggestion
            if (!subject || !message || !participantsData) {
             return;
            }
            // ... your code goes here
```